### PR TITLE
Add ROI estimate model

### PR DESCRIPTION
**Pull request by mt4110**
## Summary

- Add a vendor-neutral ROI model for Zero Image-Processing Backend pilots.
- Add a local `npm run roi:estimate` calculator with JSON and Markdown output.
- Track measured inputs separately from assumptions, confidence, caveats, and cost breakdowns.
- Add tests for CLI parsing, boundary validation, confidence levels, and tiny-value formatting.
- Register the ROI estimate claim in the claim ledger.

## Why

The cost-reduction narrative needs a measurable proof path instead of a guaranteed-savings claim. This keeps provider rates, invoices, and transformation-unit definitions as app-owned inputs while giving pilots a repeatable way to compare raw bytes, prepared bytes, backend work, and cost assumptions.

## Validation

- `npm test -- tests/docs/roi-calculator.test.mjs`
- `npm run verify`
- `npm run publish:check`

### CODE REVIEW

**Comment by mt4110** [comment]
@codex review

File: docs/tools/roi-calculator.mjs [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+      'Serverless cost per ms was supplied, but averageServerProcessingMs is zero, so processing savings are zero.'
+    );
+  }
+
+  if (
+    input.transformationCostPerUnit !== undefined &&
+    input.averageTransformationUnitsPerUpload === 1
+  ) {
+    caveats.push(
+      'Transformation savings assume one transformation unit per upload unless averageTransformationUnitsPerUpload is supplied.'
```
[the remark this answers is not in the document]
Comment (mt4110) [reply docs/tools/roi-calculator.mjs:281]: Fixed in 0e61257. The caveat now checks whether averageTransformationUnitsPerUpload was present in the original input instead of relying on the normalized default, and the explicit 1-unit case is covered by a regression test.

File: docs/tools/roi-calculator.mjs [hunk 2]
```diff
@@ -0,0 +1,613 @@
...
+    new Set(
+      fields
+        .map((field) => field.trim())
+        .filter(Boolean)
+        .map(
+          (field) =>
+            measuredAliases.get(field) ??
+            measuredAliases.get(camelize(field)) ??
+            camelize(field)
+        )
```
[the remark this answers is not in the document]
Comment (mt4110) [reply docs/tools/roi-calculator.mjs:122]: Fixed in 0e61257. normalizeMeasuredFields now validates that every measuredFields entry is a string before trimming, with a controlled error and test coverage for non-string entries.

File: docs/tools/roi-calculator.mjs [hunk 3]
```diff
@@ -0,0 +1,613 @@
...
+  assertMeasuredInputsHaveValues(input, measuredInputs);
+  const monthlyUploads = normalized.monthlyUploads;
+  const rawUploadBackendFraction = normalized.rawUploadBackendFraction;
+  const rawGbPerMonth = bytesToGb(monthlyUploads * normalized.averageRawBytes);
+  const preparedGbPerMonth = bytesToGb(
+    monthlyUploads * normalized.averagePreparedBytes
+  );
+  const avoidableRawGb = rawGbPerMonth * rawUploadBackendFraction;
+  const avoidablePreparedGb = preparedGbPerMonth * rawUploadBackendFraction;
+  const avoidedGbPerMonth = Math.max(0, avoidableRawGb - avoidablePreparedGb);
```
[the remark this answers is not in the document]
Comment (mt4110) [reply docs/tools/roi-calculator.mjs]: Fixed in 0e61257. avoidedGbPerMonth is no longer clamped; negative byte deltas now reduce bandwidth/storage savings, Markdown shows negative savings clearly, and docs/tests cover the larger-prepared-payload case.

**Comment by mt4110** [comment]
@codex review

File: docs/tools/roi-calculator.mjs [hunk 4]
```diff
@@ -0,0 +1,629 @@
...
+  );
+}
+
+function roundNumber(value, digits = 9) {
+  if (!Number.isFinite(value)) {
+    return value;
+  }
+
+  const scale = 10 ** digits;
+  const rounded = Math.round((value + Number.EPSILON) * scale) / scale;
```
[the remark this answers is not in the document]
Comment (mt4110) [reply docs/tools/roi-calculator.mjs]: Fixed in b0d9f68. roundNumber now rounds by absolute value with the original sign restored, so negative half-cent values are rounded symmetrically instead of being biased toward zero. Added a regression test for negative half-cent ROI totals.

**Comment by mt4110** [comment]
@codex review
